blog/2024-05-25-bring-slack.md

```diff
@@ -0,0 +1,266 @@
+
+```js
+let counter = new cloud.Counter();
+
+new cloud.Function(inflight () => {
+  let i = counter.inc();
+  inbox.put("file-{i}.txt", "Hello, Slack!");
+}) as "Add File";
+```
+
+Once you save the file, the Wing Console will hot reload and you should now see a function resource we can play with that looks like this:
+
+![Add File Function](assets/2024-05-25-bring-slack/add-file-function.png)
+
+So now we can click on the `Add File` function and interact with it in the right side panel. Go ahead and invoke the function a few times.
+
+![Invoke Function](assets/2024-05-25-bring-slack/invoke-function.png)
+
+And BAM!! You should now be seeing messages in your slack channel every time you invoke the function!
+
+![Slack Messages](assets/2024-05-25-bring-slack/slack-messages.png)
+
+
+## Enabling Events
+
+One thing we will notice is the slack application is supposed to support the ability to list the files in the inbox. This is done by mentioning the slack app and saying `list inbox`. 
+
+If you try this now you will see absolutely nothing happens :) this is because we need to enable events in our slack app.
+
+To do this head over to the `Event Subscriptions` section in the slack API dashboard and enable events. You will need to provide a URL for the slack API to send events to. Luckily Wing makes providing this URL easy with builtin support for tunneling.
+
+To get the tunnel URL go back to the Wing Console and `Open a tunnel for this endpoint`
+
+![Open Tunnel](assets/2024-05-25-bring-slack/open-tunnel.png)
+
+After a moment the icon will change to a eye with a slash through it, now we can copy the url by right clicking the endpoint and selecting `Copy URL`
+
+Next lets head over to the `Event Subscriptions` section in the slack API dashboard and paste the URL in the `Request URL` field. However we need to append `slack/events` to the end of the URL. So it should look something like this:
+
+![Request URL](assets/2024-05-25-bring-slack/events-sub-url.png)
+
+This should only take a few seconds to verify, and once its verified you can scroll down to the `Subscribe to Bot Events` section and add the `app_mention` event like so:
+
+![Subscribe to Bot Events](assets/2024-05-25-bring-slack/event-subscription.png)
+
+Lastly, don't forget to save your changes!
+
+### Testing the event subscription
+
+Now head back to your Slack channel where you received the messages earlier and mention your app and say `list inbox`. Our Slack apps may have different names so it wont be exactly the same as the example below:
+
+![List Inbox](assets/2024-05-25-bring-slack/list-inbox.png)
+
+You should now see a message in the channel with the files in the inbox!
+
+
+Awesome! You have now built a Slack app using Wing and tested it locally. Now lets deploy it to AWS!
+
+## Deploying to AWS
+Before we start you will need the following to follow along:
+1. An AWS Account
+2. AWS CLI installed and configured with the necessary permissions.
+3. Terraform installed
```

Comment:
Maybe we can see if the example works with OpenTofu

```suggestion
3. Terraform or OpenTofu installed
```

Reply:
Good idea, Ill try it out